source/agora/node/Ledger.d

```diff
@@ -1352,3 +1354,143 @@ unittest
     genBlockTransactionsFreeze(1, TxType.Payment);
     assert(ledger.getBlockHeight() == 11 + 2016);
 }
+
+/// test enrollments in the genesis block
+unittest
+{
+    import agora.common.BitField;
+    import agora.common.Serializer;
+    import std.exception;
+
+    static class PreLoadedMemBlockStorage : MemBlockStorage
+    {
+        const(Block)[] blocks;
+
+        public this (const(Block)[] blocks)
+        {
+            this.blocks = blocks;
+        }
+
+        public override bool load ()
+        {
+            foreach (const ref block; this.blocks)
+                this.saveBlock(block);
+
+            return true;
+        }
+    }
+
+    // generate genesis with a freeze & payment tx, and 'count' number of
+    // extra blocks
+    const(Block)[] genBlocksToIndex ( KeyPair key_pair,
+        EnrollmentManager enroll_man, size_t count)
+    {
+        // 1 payment and 1 freeze tx (must be a power of 2 due to #797)
+        Transaction[] gen_txs;
+        // need mutable
+        gen_txs ~= GenesisTransaction().serializeFull.deserializeFull!Transaction;
+
+        Transaction freeze_tx =
+        {
+            type : TxType.Freeze,
+            inputs : [ Input.init ],
+            outputs : [Output(Amount.MinFreezeAmount, key_pair.address)]
+        };
+
+        gen_txs ~= freeze_tx;
+        Hash txhash = hashFull(freeze_tx);
+        Hash utxo = UTXOSet.getHash(txhash, 0);
+
+        Enrollment[] enrolls;
+        Enrollment enroll;
+        const StartHeight = 0;  // not important
+        assert(enroll_man.createEnrollment(utxo, StartHeight, enroll));
+        enrolls ~= enroll;
+
+        gen_txs.sort;
+        Hash[] merkle_tree;
+        auto merkle_root = Block.buildMerkleTree(gen_txs, merkle_tree);
+
+        auto genesis = immutable(Block)(
+            immutable(BlockHeader)(
+                Hash.init,   // prev
+                0,           // height
+                merkle_root,
+                BitField!uint.init,
+                Signature.init,
+                enrolls.assumeUnique,
+            ),
+            gen_txs.assumeUnique,
+            merkle_tree.assumeUnique
+        );
+
+        const(Block)[] blocks = [genesis];
+
+        const(Transaction)[] prev_txs;
+        foreach (_; 0 .. count)
+        {
+            auto txs = makeChainedTransactions(getGenesisKeyPair(),
+                prev_txs, 1);
+
+            const NoEnrollments = null;
+            blocks ~= makeNewBlock(blocks[$ - 1], txs, NoEnrollments);
+            prev_txs = txs;
+        }
+
+        return blocks.assumeUnique;
+    }
+
+    // only genesis loaded: validator is active
+    {
```

Comment:
@Geod24 if you don't like these `{ }` scopes I'll move them to separate tests. I just wanted to avoid accidentally re-using existing variables. Re-using the enrollment manager for example would cause a bug here.

Reply:
No that works great to split tests